tests/cmds/test_devices.py

```diff
@@ -0,0 +1,478 @@
+    response.text = text
+    response._content_consumed = mocker.MagicMock()
+    response.status_code = 200
+    return Py42Response(response)
+
+
+@pytest.fixture
+def mock_device_settings(mocker):
+    device_settings = mocker.MagicMock()
+    device_settings.name = "testname"
+    return device_settings
+
+
+@pytest.fixture
+def deactivate_response(mocker):
+    return _create_py42_response(mocker, "")
+
+
+@pytest.fixture
+def device_info_response(mocker):
+    return _create_py42_response(mocker, TEST_DEVICE_RESPONSE)
+
+
+@pytest.fixture
+def archives_list_generator(mocker):
+    yield TEST_ARCHIVES_RESPONSE
+
+
+@pytest.fixture
+def devices_list_generator(mocker):
+    return [TEST_COMPUTER_PAGE]
+
+
+@pytest.fixture
+def users_list_generator(mocker):
+    return [TEST_USERS_LIST_PAGE]
+
+
+@pytest.fixture
+def backupusage_response(mocker):
+    return _create_py42_response(mocker, TEST_BACKUPUSAGE_RESPONSE)
+
+
+@pytest.fixture
+def empty_backupusage_response(mocker):
+    return _create_py42_response(mocker, TEST_EMPTY_BACKUPUSAGE_RESPONSE)
+
+
+@pytest.fixture
+def device_info_success(cli_state, device_info_response):
+    cli_state.sdk.devices.get_by_id.return_value = device_info_response
+
+
+@pytest.fixture
+def archives_list_success(cli_state, archives_list_generator):
+    cli_state.sdk.archive.get_all_by_device_guid.return_value = archives_list_generator
+
+
+@pytest.fixture
+def deactivate_device_success(cli_state, deactivate_response):
+    cli_state.sdk.devices.deactivate.return_value = deactivate_response
+
+
+@pytest.fixture
+def deactivate_device_not_found_failure(cli_state):
+    cli_state.sdk.devices.deactivate.side_effect = Py42NotFoundError(HTTPError())
+
+
+@pytest.fixture
+def deactivate_device_in_legal_hold_failure(cli_state):
+    cli_state.sdk.devices.deactivate.side_effect = Py42BadRequestError(HTTPError())
+
+
+@pytest.fixture
+def deactivate_device_not_allowed_failure(cli_state):
+    cli_state.sdk.devices.deactivate.side_effect = Py42ForbiddenError(HTTPError())
+
+
+@pytest.fixture
+def backupusage_success(cli_state, backupusage_response):
+    cli_state.sdk.devices.get_by_id.return_value = backupusage_response
+
+
+@pytest.fixture
+def empty_backupusage_success(cli_state, empty_backupusage_response):
+    cli_state.sdk.devices.get_by_id.return_value = empty_backupusage_response
+
+
+@pytest.fixture
+def get_all_devices_success(cli_state, devices_list_generator):
+    cli_state.sdk.devices.get_all.return_value = devices_list_generator
+
+
+@pytest.fixture
+def get_all_users_success(cli_state, users_list_generator):
+    cli_state.sdk.users.get_all.return_value = users_list_generator
+
+
+def test_deactivate_deactivates_device(runner, cli_state, deactivate_device_success):
+    runner.invoke(
+        cli, ["devices", "deactivate", "--device-id", TEST_DEVICE_ID], obj=cli_state
+    )
+    cli_state.sdk.devices.deactivate.assert_called_once_with(TEST_DEVICE_ID)
+
+
+def test_deactivate_updates_purge_date(
```

Comment:
Like the test below, should have the state as part of the signature, like `test_deactivate_when_given_purge_flag_purges_date()` or else it sounds like a default behavior test